### PR TITLE
Added schema validation when creating a server method using object

### PR DESCRIPTION
**Pull request by mtharrison**
This is designed to catch simple mistakes like this:

    server.method({
        name: 'myMethod',
        method: function () { ... },
        cache: {
            expiresIn: 1000
        }
    });

which should really be:

    server.method({
        name: 'myMethod',
        method: function () { ... },
        options: {
            cache: {
                expiresIn: 1000
            }
        }
    });

### CODE REVIEW

File: lib/schema.js [hunk 1]
```diff
@@ -274,6 +274,13 @@ internals.method = Joi.object({
 });
 
 
+internals.methodObject = Joi.object({
+    name: Joi.string(),
```
Comment (Marsup) [inline lib/schema.js]: Shouldn't name and method be required ?

Comment (mtharrison) [inline lib/schema.js]: Yes. At the moment that's still taken care of here: 

https://github.com/hapijs/hapi/blob/master/lib/methods.js#L42-L43

But I can add it to this schema too to be more explicit about it.

**Comment by AdriVanHoudt** [comment]
:+1:

**Comment by netei** [comment]
Could be usefull indead

**Pull request merged**